core/file_test.go

```diff
@@ -0,0 +1,239 @@
+package core_test
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go/config"
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/logger"
+	"github.com/stretchr/testify/assert"
+)
+
+type TestStruct struct {
+	a, b int
+}
+
+func TestOpenFile_NoExistingFileShouldErr(t *testing.T) {
+	t.Parallel()
+
+	file, err := core.OpenFile("testFile1", logger.DefaultLogger())
+
+	assert.Nil(t, file)
+	assert.Error(t, err)
+}
+
+func TestOpenFile_NoErrShouldPass(t *testing.T) {
+	t.Parallel()
+
+	_, err := os.Create("testFile2")
+
+	assert.Nil(t, err)
+
+	file, err := core.OpenFile("testFile2", logger.DefaultLogger())
+
+	if _, errF := os.Stat("testFile2"); errF == nil {
+		_ = os.Remove("testFile2")
+	}
+
+	assert.NotNil(t, file)
+	assert.Nil(t, err)
+}
+
+func TestLoadTomlFile_NoExistingFileShouldErr(t *testing.T) {
+	t.Parallel()
+
+	cfg := &config.Config{}
+
+	err := core.LoadTomlFile(cfg, "file", logger.DefaultLogger())
+
+	assert.Error(t, err)
+}
+
+func TestLoadTomlFile_FileExitsShouldPass(t *testing.T) {
+	t.Parallel()
+
+	cfg := &config.Config{}
+
+	_, err := os.Create("testFile3")
+
+	assert.Nil(t, err)
+
+	err = core.LoadTomlFile(cfg, "testFile3", logger.DefaultLogger())
+
+	if _, errF := os.Stat("testFile3"); errF == nil {
+		_ = os.Remove("testFile3")
+	}
+
+	assert.Nil(t, err)
+}
+
+func TestLoadJSonFile_NoExistingFileShouldErr(t *testing.T) {
+	t.Parallel()
+
+	cfg := &config.Config{}
+
+	err := core.LoadJsonFile(cfg, "file", logger.DefaultLogger())
+
+	assert.Error(t, err)
+
+}
+
+func TestLoadJSonFile_FileExitsShouldPass(t *testing.T) {
+	t.Parallel()
+
+	cfg := &config.Config{}
+
+	file, err := os.Create("testFile4")
+
+	assert.Nil(t, err)
+
+	data, _ := json.MarshalIndent(TestStruct{}, "", " ")
+
+	_ = ioutil.WriteFile("testFile4", data, 0644)
```

Comment:
use a variable instead of "testFile4" do the same for all the tests where the testFile is used multiple times.

Reply:
done